### DISCUSSION TITLE
Range implementing IEnumerable<int>

### DISCUSSION THREAD

**Discussion by lucasteles** [dotnet/runtime#68441] [General]
I think would be great to have a simpler way to define a collection range outside `Enumerable.Range`, which is not from-to, It is from and size.

A way to do that today is using the `Range` data type.

```csharp
foreach (var n in 1..10)
    Console.WriteLine(n);

var things = (1..10)
            .AsEnumerable()
            .Select(GetStuff)
            .Where(x => x.Length > 0);

public static class RangeExtensions {
    public static IEnumerator<int> GetEnumerator(this Range range) => 
        range.AsEnumerable().GetEnumerator();
    public static IEnumerable<int> AsEnumerable(this Range range) => 
        Enumerable.Range(range.Start.Value, range.End.Value - range.Start.Value);
}
```

If `Range` implements IEnumerable this could work nice, dropping the use of `.AsEnumerable()` and gaining a cool syntax to define rage values

**Comment by CyrusNajmabadi** [comment]
Transferring to runtime.

**Comment by hez2010** [comment]
Duplicates #462
